### ISSUE TITLE
"notify" field name does not compile

### ISSUE THREAD

**Issue by martinreilent** [ermadmi78/kobby#42]
STR:
1. Kobby  plugin section in pom.xml
```xml
<plugin>
  <groupId>io.github.ermadmi78</groupId>
  <artifactId>kobby-maven-plugin</artifactId>
  <version>4.0.1</version>
  <configuration>
      <schema>
          <scan>
              <includes>
                  <include>**/*.graphqls</include>
                  <include>**/*.graphql</include>
              </includes>
          </scan>
      </schema>
      <kotlin>
          <packageName>generated</packageName>
          <outputDirectory>src/main/kotlin</outputDirectory>
      </kotlin>
  </configuration>
</plugin>
```
2. Schema 
```graphql
type Mutation {
  editThing( notify: Boolean! ): Thing
}

type Thing {
  field: String!
  field2: Int!
  notify: Boolean!
}
```
3. Run
```bash
mvn clean kobby:generate-kotlin && mvn clean compile
```

**Expected**
No errors

**Actual**
[ERROR] .../src/main/kotlin/generated/entity/Thing.kt: (44, 3) Accidental override: The following declarations have the same JVM signature (notify()V):
    fun notify(): Unit defined in generated.entity.ThingProjection
[ERROR] .../src/main/kotlin/generated/entity/impl/ThingImpl.kt: (116, 3) Accidental override: The following declarations have the same JVM signature (notify()V):
    fun notify(): Unit defined in generated.entity.impl.ThingProjectionImpl
 
-----

Problem seems to be with creating projections of fields that are with same name as java.lang.Object functions. 
Possible workaround currently is to prefix all projection fields 

```xml
...
<!-- Prefix for projection fields -->
<!-- that are not marked with the directive `@default` -->
<withPrefix></withPrefix>
...
```

but this looks ugly in the end.

**Comment by ermadmi78** [comment]
I think that is best to fix in the same style as here #21

**Comment by ermadmi78** [comment]
In addition, it will be necessary to correct all the names of the methods java.lang.Object

**Comment by ermadmi78** [comment]
The temporary workaround is to use `kotlin.entity.projection.withPrefix` or `kotlin.entity.projection.withPostfix` plugin settings.

Gradle example:
```kotlin
plugins {
    id("io.github.ermadmi78.kobby") version "4.0.1"
}

kobby {
    kotlin {
        entity {
            projection {
                withPrefix = "with"
            }
        }
    }
}
```

Maven example:
```xml
<plugin>
     <groupId>io.github.ermadmi78</groupId>
     <artifactId>kobby-maven-plugin</artifactId>
     <version>4.0.1</version>
     <executions>
         <execution>
             <phase>generate-sources</phase>
             <goals>
                 <goal>generate-kotlin</goal>
             </goals>
             <configuration>
                 <kotlin>
                     <entity>
                         <projection>
                             <withPrefix>with</withPrefix>
                         </projection>
                     </entity>
                 </kotlin>
             </configuration>
         </execution>
     </executions>
 </plugin>
```

This plugin setting will add "with" prefix to all generated projection methods.

**Comment by ermadmi78** [comment]
The problem is reproduced for all names that match `java.lang.Object` [method names](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Object.html):
- `getClass`
- `hashCode`
- `equals`
- `clone`
- `toString`
- `notify`
- `notifyAll`
- `wait`
- `finalize`

To fix this problem, Kobby will automatically add prefix "__" to all forbidden projection names.

For example, for type:
```graphql
type Query {
    getClass: Boolean!
    hashCode: Boolean!
    equals: Boolean!
    clone: Boolean!
    toString: Boolean!
    notify: Boolean!
    notifyAll: Boolean!
    wait: Boolean!
    finalize: Boolean!

    normal: Boolean!
}
```

Kobby will generate projection:
```kotlin
public interface QueryProjection {
  public fun __getClass(): Unit

  public fun __hashCode(): Unit

  public fun __equals(): Unit

  public fun __clone(): Unit

  public fun __toString(): Unit

  public fun __notify(): Unit

  public fun __notifyAll(): Unit

  public fun __wait(): Unit

  public fun __finalize(): Unit

  public fun normal(): Unit

  public fun __minimize(): Unit {
  }
}
```

**Comment by ermadmi78** [comment]
Fixed in release 4.0.3
